caluma/form/validators.py

```diff
@@ -0,0 +1,94 @@
+import sys
+
+from rest_framework import exceptions
+
+from . import jexl
+
+
+class AnswerValidator:
+    def _validate_question_text(self, question, value):
+        max_length = (
+            question.max_length if question.max_length is not None else sys.maxsize
+        )
+        if not isinstance(value, str) or len(value) > max_length:
```

Comment:
Doesn't Django have built-in validators for such cases?
See https://docs.djangoproject.com/en/2.1/ref/validators/#built-in-validators

Reply:
It would be but those validators are per field and do not have access to the full context of the model.

If we wanted to use those we would have to normalize the database to have  one column or actually even table for each value type. Makes the code and database model more complicated so we decided to simplify this on this part and use a `JSONField` for values. Hence this validator is which has access to the question as well.